src/Elastic.Transport/Diagnostics/TcpStats.cs

```diff
@@ -29,8 +29,19 @@ public static class TcpStats
 		public static ReadOnlyDictionary<TcpState, int> GetStates()
 		{
 			var states = new Dictionary<TcpState, int>(StateLength);
-			var connections = GetActiveTcpConnections();
-			for (var index = 0; index < connections.Length; index++)
+
+			TcpConnectionInformation[] connections = null;
+
+			try
+			{
+				connections = GetActiveTcpConnections();
```

Comment:
since `GetActiveTcpConnections()` is a public method, I think the `try/catch` should be inside `GetActiveTcpConnections()`.

I think it ought to return `null` as you propose here, rather than `Array.Empty<TcpConnectionInformation>`, to imply that the information could not be retrieved rather than there were no connections

Reply:
Agreed! Will update now..